worldedit-core/src/main/java/com/fastasyncworldedit/core/configuration/Settings.java

```diff
@@ -622,6 +622,13 @@ public static class PROGRESS {
     })
     public static class EXPERIMENTAL {
 
+        @Comment({
+                "Undo operation batch size",
+                " - The size defines the number of changes read at once.",
+                " - Larger numbers might reduce overhead but increase latency for edits with only few changes.",
+                " - 0 means undo operations are not batched."})
+        public int UNDO_BATCH_SIZE = 128;
```

Comment:
We should know history size before undoing, so we could dynamically set this?

Reply:
I'm not sure if dynamically setting it has much benefit, it's basically just a tradeoff between cross-thread-communication and memory overhead. Not sure if there is any good way to find an appropriate value for it.

Reply:
I suppose if people care that much they can find the best value for themselves, otherwise, I would assume very large edits where this would matter more is going to be more affected by chunk loading times anyway